core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -17,11 +21,13 @@
 /**
  * Encloses classes related to the compression and decompression of messages.
  */
+@ExperimentalApi
 public final class MessageEncoding {
   /**
    * Special sentinel codec indicating that no compression should be used.  Users should use
    * reference equality to see if compression is disabled.
    */
+  @ExperimentalApi
```

Comment:
ChannelImpl, ClientCallImpl, MessageDeframer, and MessageDeframer.  They use a mix of != and ==, in order to be able to use null as well.